src/result_processor.c

```diff
@@ -560,7 +560,12 @@ static int rploaderNext(ResultProcessor *base, SearchResult *r) {
   } else {
     loadopts.mode |= RLOOKUP_LOAD_ALLKEYS;
   }
-  RLookup_LoadDocument(lc->lk, &r->rowdata, &loadopts);
+
```

Comment:
Where do you call the next again?

Reply:
Had other confusion with the flow
See the latest. 
This is indeed loading before the reply starts.

Reply:
Don't understand how it works. you must call the next again to get another result no?